cpuinfo.go

```diff
@@ -190,15 +190,26 @@ func parseCPUInfoARM(info []byte) ([]CPUInfo, error) {
 	scanner := bufio.NewScanner(bytes.NewReader(info))
 
 	firstLine := firstNonEmptyLine(scanner)
-	if !strings.HasPrefix(firstLine, "Processor") || !strings.Contains(firstLine, ":") {
+	match, _ := regexp.MatchString("^[Pp]rocessor", firstLine)
+	if !match || !strings.Contains(firstLine, ":") {
 		return nil, errors.New("invalid cpuinfo file: " + firstLine)
 	}
 	field := strings.SplitN(firstLine, ": ", 2)
-	commonCPUInfo := CPUInfo{VendorID: field[1]}
-
 	cpuinfo := []CPUInfo{}
-	i := -1
 	featuresLine := ""
+	commonCPUInfo := CPUInfo{}
+	i := 0
+	if field[0] == "Processor " {
+		commonCPUInfo = CPUInfo{VendorID: field[1]}
+		i = -1
```

Comment:
Sadly the format appears to be a bit more complicated. If you look at the [commit](https://github.com/torvalds/linux/commit/b4b8f770eb10a1bccaf8aa0ec1956e2dd7ed1e0a) you see that the presence of "processor" was conditional on CONFIG_SMP. In other words, for cases where you only have one core it's omitted.

I found this example of an old cpuinfo from a Raspberry Pi:

```
pelicone ~ # cat /proc/cpuinfo
model name : ARMv6-compatible processor rev 7 (v6l)
Features : half thumb fastmult vfp edsp java tls
CPU implementer : 0x41
CPU architecture: 7
CPU variant : 0x0
CPU part : 0xb76
CPU revision : 7

Hardware : BCM2708
Revision : 0010 
```

Reply:
Didn't check the 'legacy' format, just focused on the newer one so I totally missed this. I'm going to add this example to the test cases and ensure the code also supports it. Thanks!